tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptioninvoiceaccount/SubscriptionInvoiceAccountCreatePrepaymentTest.java

```diff
@@ -0,0 +1,124 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SubscriptionInvoiceAccountCreatePrepaymentTest {
+    private static final TestSetup TEST_SETUP = new TestSetup();
+    private static final AdvancedBillingClient CLIENT = TestClient.createClient();
+    private static final SubscriptionInvoiceAccountController SUBSCRIPTION_INVOICE_ACCOUNT_CONTROLLER =
+            CLIENT.getSubscriptionInvoiceAccountController();
+
+    private static Customer customer;
+    private static Subscription subscription;
+
+    @BeforeAll
+    static void setUp() throws IOException, ApiException {
+        customer = TEST_SETUP.createCustomer();
+        subscription = TEST_SETUP.createSubscription(customer, TEST_SETUP.createProduct(
+                        TEST_SETUP.createProductFamily(),
+                        b -> b.priceInCents(1250)
+                )
+        );
+    }
+
+    @AfterAll
+    static void teardown() throws IOException, ApiException {
+        new TestTeardown().deleteCustomer(customer);
+    }
+
+    @Test
+    void shouldCreatePrepaymentForSubscription() throws IOException, ApiException {
+        // when
+        CreatedPrepayment prepayment = SUBSCRIPTION_INVOICE_ACCOUNT_CONTROLLER
+                .createPrepayment(
+                        subscription.getId(),
+                        new CreatePrepaymentRequest(new CreatePrepayment.Builder()
+                                .amount(100.0)
+                                .details("John Doe signup for $100")
+                                .memo("Signup for $100")
+                                .method(CreatePrepaymentMethod.CHECK)
+                                .build()
+                        )
+                )
+                .getPrepayment();
+
+        // then
+        assertThat(prepayment.getAdditionalProperties()).isEmpty();
+        assertThat(prepayment.getId()).isNotNull();
+        assertThat(prepayment.getSubscriptionId()).isEqualTo(subscription.getId());
+        assertThat(prepayment.getAmountInCents()).isEqualTo(10000L);
+        assertThat(prepayment.getMemo()).isEqualTo("Signup for $100");
+        assertThat(prepayment.getCreatedAt()).isBeforeOrEqualTo(ZonedDateTime.now());
+        assertThat(prepayment.getStartingBalanceInCents()).isEqualTo(0L);
+        assertThat(prepayment.getEndingBalanceInCents()).isEqualTo(-10000L);
+    }
+
+    @ParameterizedTest
+    @MethodSource("argsForShouldReturn422WhenRequestBodyIsIncorrect")
+    void shouldReturn422WhenRequestBodyIsIncorrect(CreatePrepaymentRequest request, String expectedErrors) {
+        // when - then
+        assertUnprocessableEntity(
+                ApiException.class,
+                () -> SUBSCRIPTION_INVOICE_ACCOUNT_CONTROLLER.createPrepayment(subscription.getId(), request),
+                e -> JSONAssert.assertEquals(expectedErrors, e.getHttpContext().getResponse().getBody(), JSONCompareMode.STRICT)
```

Comment:
Do we need to assert using Json? Or can we use exception class here?

Reply:
Unfortunately, this is the only solution (we already have other places using that approach) because the generated SDK doesn't support `oneOf` for error schemas and only the generic one is thrown.